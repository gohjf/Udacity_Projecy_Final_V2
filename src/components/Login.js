import { useState } from "react";
import { connect } from "react-redux";
import { handleLoginUser } from "../actions/authedUser.js";
import { Navigate, useLocation } from "react-router-dom";

const Login = (props) => {
    const [userName, setUserName] = useState("");
    const [passWord, setPassWord] = useState("");
    const location = useLocation();
    //const urlParams = new URLSearchParams(window.location.search);
    const redirectUrl = location.state?.from || "/";
    //const redirectUrl = urlParams.get('redirectTo');

    if (props.isAuthorised === true) {
        console.log(redirectUrl);
        return <Navigate to={redirectUrl ? redirectUrl : "/"} />;
    }

    const handleUserNameChange = (event) => {
        const value = event.target.value;
        setUserName(value);
    };

    const handlePassWordChange = (event) => {
        const value = event.target.value;
        setPassWord(value);
    };

    const handleSubmit = (event) => {
        event.preventDefault();
        props.dispatch(handleLoginUser(userName, passWord));
        setUserName("");
        setPassWord("");
    };


    return (
        <div>
            <form onSubmit={handleSubmit} >
                <div className="LoginHeader">
                    <h3>Employee Polls</h3>
                </div>
                <div className="LoginTitle">
                    <h3>
                        Login:
                    </h3>
                </div>
                <div className="LoginInput">
                    <textarea
                        className="LoginUserName"
                        placeholder="UserName"
                        value={userName}
                        onChange={handleUserNameChange}
                        data-testid={"username_login"}
                    >
                    </textarea>
                </div>
                <div className="PasswordInput">
                    <textarea
                        className="LoginPassword"
                        placeholder="Password"
                        value={passWord}
                        onChange={handlePassWordChange}
                        data-testid={"Password_login"}
                    >
                    </textarea>
                </div>
                <div className="SubmitButton">
                    <button
                        disabled={userName === "" || passWord === ""}
                        type="submit"
                        data-testid={"Button_login"}>
                        Login
                    </button>
                </div>
            </form>
        </div>
    )

};

const mapStateToProps = ({ authedUser, users }) => {
    return {
        isAuthorised: authedUser === null ? false : true,
        users: Object.values(users),
    };
};

export default connect(mapStateToProps)(Login);