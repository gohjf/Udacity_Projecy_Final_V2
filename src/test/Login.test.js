import Login from '../components/Login';
import { render } from '@testing-library/react';
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { store } from "../store.js";
import * as React from "react";
import '@testing-library/jest-dom'

describe('Login', () => {
    it('will match snapshot', () => {
        var Component = render(
            <Provider store={store}>
            <Router>
              <Login />
            </Router>
          </Provider>);
    
    expect(Component).toMatchSnapshot();
    })
}) 

describe('Login', () => {
    it('will have all expected fields', () => {
        var Component = render(
            <Provider store={store}>
            <Router>
              <Login />
            </Router>
          </Provider>);

        var loginInput = Component.getByTestId('username_login');
        var passwordInput = Component.getByTestId('Password_login');
        var SubmitButton = Component.getByTestId('Button_login');

        expect(loginInput).toBeInTheDocument();;
        expect(passwordInput).toBeInTheDocument();
        expect(SubmitButton).toBeInTheDocument();
    })
}) 