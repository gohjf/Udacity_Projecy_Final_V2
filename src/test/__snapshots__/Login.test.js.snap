// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login will have all expected fields 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <div
            class="LoginHeader"
          >
            <h3>
              Employee Polls
            </h3>
          </div>
          <div
            class="LoginTitle"
          >
            <h3>
              Login:
            </h3>
          </div>
          <div
            class="LoginInput"
          >
            <textarea
              class="LoginUserName"
              data-testid="username_login"
              placeholder="UserName"
            />
          </div>
          <div
            class="PasswordInput"
          >
            <textarea
              class="LoginPassword"
              data-testid="Password_login"
              placeholder="Password"
            />
          </div>
          <div
            class="SubmitButton"
          >
            <button
              data-testid="Button_login"
              disabled=""
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <div
          class="LoginHeader"
        >
          <h3>
            Employee Polls
          </h3>
        </div>
        <div
          class="LoginTitle"
        >
          <h3>
            Login:
          </h3>
        </div>
        <div
          class="LoginInput"
        >
          <textarea
            class="LoginUserName"
            data-testid="username_login"
            placeholder="UserName"
          />
        </div>
        <div
          class="PasswordInput"
        >
          <textarea
            class="LoginPassword"
            data-testid="Password_login"
            placeholder="Password"
          />
        </div>
        <div
          class="SubmitButton"
        >
          <button
            data-testid="Button_login"
            disabled=""
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <div
            class="LoginHeader"
          >
            <h3>
              Employee Polls
            </h3>
          </div>
          <div
            class="LoginTitle"
          >
            <h3>
              Login:
            </h3>
          </div>
          <div
            class="LoginInput"
          >
            <textarea
              class="LoginUserName"
              data-testid="username_login"
              placeholder="UserName"
            />
          </div>
          <div
            class="PasswordInput"
          >
            <textarea
              class="LoginPassword"
              data-testid="Password_login"
              placeholder="Password"
            />
          </div>
          <div
            class="SubmitButton"
          >
            <button
              data-testid="Button_login"
              disabled=""
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <div
          class="LoginHeader"
        >
          <h3>
            Employee Polls
          </h3>
        </div>
        <div
          class="LoginTitle"
        >
          <h3>
            Login:
          </h3>
        </div>
        <div
          class="LoginInput"
        >
          <textarea
            class="LoginUserName"
            data-testid="username_login"
            placeholder="UserName"
          />
        </div>
        <div
          class="PasswordInput"
        >
          <textarea
            class="LoginPassword"
            data-testid="Password_login"
            placeholder="Password"
          />
        </div>
        <div
          class="SubmitButton"
        >
          <button
            data-testid="Button_login"
            disabled=""
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
